samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/delivery/SubmitToGradingActionListener.java

```diff
@@ -657,7 +657,13 @@ private void prepareItemGradingPerItem(ActionEvent ae, DeliveryBean delivery,
 							|| itemgrading.getAttemptDate() != null) { 
 							// null=> skipping this question
 							itemgrading.setAgentId(AgentFacade.getAgentString());
-							itemgrading.setSubmittedDate(new Date());
+							int currentTimeInSeconds = 0;
+							if (itemgrading.getAttemptDate() != null) {
+								currentTimeInSeconds = (int) ((new Date()).getTime()/1000 - itemgrading.getAttemptDate().getTime()/1000);
+							}
+							if (item.getTimeLimit().equals("false") || Integer.parseInt(item.getTimeLimit()) >= currentTimeInSeconds) {
+								itemgrading.setSubmittedDate(new Date());
+							}
```

Comment:
It looks this exact code is being repeated a few times, what do you think of creating a method so we can save lines? Also plus it will be easy to maintain: 1 place to modify it vs X appearances.